devito/builtins/arithmetic.py

```diff
@@ -58,6 +57,58 @@ def norm(f, order=2):
     return f.dtype(v)
 
 
+@dv.switchconfig(log_level='ERROR')
+def sum(f, dims=None):
+    """
+    Compute the sum of the Function data over specified dimensions.
+    Defaults to sum over all dmensions
+
+    Parameters
+    ----------
+    f : Function
+        Input Function.
+    dims : Dimension or tuple of Dimension
+        Dimensions to sum over.
+    """
+    dims = dv.tools.as_tuple(dims)
+    if dims == () or dims == f.dimensions:
+        return sumall(f)
+
+    # Get dimensions and shape of the result.
+    new_dims = tuple(d for d in f.dimensions if d not in dims)
+    shape = tuple(f._size_domain[d] for d in new_dims)
+    if f.is_TimeFunction and f.time_dim not in dims:
+        out = dv.TimeFunction(name="%ssum" % f.name, grid=f.grid,
+                              space_order=f.space_order, shape=shape,
+                              dimensions=new_dims, time_dim=f.time_dim,
+                              time_order=f.time_order)
+    elif f.is_SparseTimeFunction:
+        if f.time_dim in dims:
+            # Sum over time -> SparseFunction
+            out = dv.SparseFunction(name="%ssum" % f.name, grid=f.grid,
+                                    dimensions=new_dims, npoint=f.shape[1],
+                                    coordinates=f.coordinates)
+        else:
+            # Sum over rec -> TimeFunction
+            out = dv.TimeFunction(name="%ssum" % f.name, grid=f.grid, shape=shape,
+                                  dimensions=new_dims, space_order=0,
+                                  time_order=f.time_order)
+    else:
+        out = dv.Function(name="%ssum" % f.name, grid=f.grid,
+                          space_order=f.space_order, shape=shape,
+                          dimensions=new_dims)
+
+    kwargs = {}
+    if f.is_TimeFunction and f._time_buffering:
+        kwargs[f.time_dim.max_name] = f._time_size - 1
+
+    # Only need one guard has they have the same coordiantes and dimensions
```

Comment:
has they have?

Reply:
has -> as

also, "Dimensions" (to be picky)